src/Airship/Resource.hs

```diff
@@ -151,3 +152,8 @@ defaultResource = Resource { allowMissingPost          = return False
                            , validContentHeaders       = return True
                            , errorResponses            = mempty
                            }
+
+
+instance Monad m => Monoid (Resource m) where
```

Comment:
This instance violates the following Monoid laws:

```
mappend mempty x = x
mappend x (mappend y z) = mappend (mappend x y) z
 -- Most importantly
mconcat = foldr mappend mempty
```
 We could instead use the Monoid instance of `First` and the function `fromMaybe defaultResource . getFirst` to achieve the same.